rclcpp/include/rclcpp/clock.hpp

```diff
@@ -194,12 +194,17 @@ class Clock
   ros_time_is_active();
 
   /**
+   * Deprecated. This API is broken, as there is no way to get a deep
+   * copy of a clock. Therefore one can experience spurious wakeups triggered
+   * by some other instance of a clock.
```

Comment:
how about adding the test cases for `ClockConditionalVariable` including this case? generally those new classes would be nice to have general unit tests too.